src/Event.cc

```diff
@@ -97,27 +97,63 @@ Event::Event() {
 void Event::_init() {
     _keywords.insert(TYPE);
     _keywords.insert(EVENTTIME);
-    _keywords.insert(HOSTID);
-    _keywords.insert(RUNID);
+//    _keywords.insert(RUNID);
     _keywords.insert(STATUS);
     _keywords.insert(TOPIC);
     _keywords.insert(PUBTIME);
 }
 
 Event::Event(cms::TextMessage *msg) {
-    _init();
+    // _init();
+
+    vector<std::string>names = msg->getPropertyNames();
+    unsigned int i;
 
     _psp = processTextMessage(msg);
 
-    _psp->set(TYPE, msg->getStringProperty(TYPE));
-    _psp->set(HOSTID, msg->getStringProperty(HOSTID));
-    _psp->set(RUNID, msg->getStringProperty(RUNID));
-    _psp->set(STATUS, msg->getStringProperty(STATUS));
-    _psp->set(TOPIC, msg->getStringProperty(TOPIC));
-    _psp->set(EVENTTIME, msg->getLongProperty(EVENTTIME));
-    _psp->set(PUBTIME, msg->getLongProperty(PUBTIME));
-/*
-*/
+    for (i = 0; i < names.size(); i++)
+            _keywords.insert(names[i]);
+
+    _psp->set(EVENTTIME, msg->getCMSTimestamp());
+
+    for (i = 0; i < names.size(); i++) {
+        std::string name = names[i];
+        cms::Message::ValueType vType = msg->getPropertyValueType(names[i]);
+        switch(vType) {
+            case cms::Message::NULL_TYPE:
+                _psp->set(name, NULL);
+                break;
+            case cms::Message::BOOLEAN_TYPE:
+                _psp->set(name, msg->getBooleanProperty(name));
+                break;
+            case cms::Message::BYTE_TYPE:
+            case cms::Message::CHAR_TYPE:
+                _psp->set(name, msg->getByteProperty(name));
+                break;
+            case cms::Message::SHORT_TYPE:
+                _psp->set(name, msg->getShortProperty(name));
+                break;
+            case cms::Message::INTEGER_TYPE:
+                _psp->set(name, msg->getIntProperty(name));
+                break;
+            case cms::Message::LONG_TYPE:
+                _psp->set(name, msg->getLongProperty(name));
+                break;
+            case cms::Message::DOUBLE_TYPE:
+                _psp->set(name, msg->getDoubleProperty(name));
+                break;
+            case cms::Message::FLOAT_TYPE:
+                _psp->set(name, msg->getFloatProperty(name));
+                break;
+            case cms::Message::STRING_TYPE:
+                _psp->set(name, msg->getStringProperty(name));
+                break;
+            case cms::Message::BYTE_ARRAY_TYPE:
+            case cms::Message::UNKNOWN_TYPE:
+                break;
```

Comment:
Silently letting these drop on the floor.  You should at the least issue a warning.
Where's the `default` case?